### ISSUE TITLE
App is crashing when sync button is clicked

### ISSUE THREAD

**Issue by mustaqmustu** [opendatakit/skunkworks-bat#9]
App is crashing when sync button is clicked

**Comment by jbeorse** [comment]
@mustaqmustu Can you provide reproduction steps?

**Comment by Saikat2019** [comment]
@jbeorse I think ODK Services should be opened when sync button will be clicked, if yes,can I work on it?

**Comment by DreamyPhobic** [comment]
Actually, there is some error in connecting the database with this app. Let me break down this problem: 

First, if ODK services app is not installed on device, then on clicking Sync button it will give ```IllegalStateException``` because FirebaseApp is not initialized in Application class.
Here the logs are:
```
E/MessageQueue-JNI: java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process org.odk.odknotifications. Make sure to call FirebaseApp.initializeApp(Context) first.
        at com.google.firebase.FirebaseApp.getInstance(SourceFile:218)
        at com.google.firebase.database.FirebaseDatabase.getInstance(Unknown Source:0)
        at org.odk.odknotifications.Activities.MainActivity.addGroupsFromFirebase(MainActivity.java:216)
        at org.odk.odknotifications.Activities.MainActivity.syncCloudDatabase(MainActivity.java:210)
        at org.odk.odknotifications.Activities.MainActivity.onOptionsItemSelected(MainActivity.java:201)
        at android.app.Activity.onMenuItemSelected(Activity.java:3600)
```

Second, if ODK services app already installed on device then it will give another error which actually hides first error. Here, the error is:
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.odk.odknotifications, PID: 14172
    java.lang.IllegalStateException: not IllegalStateException or RemoteException on AidlDbInterface: java.lang.SecurityException: java.lang.SecurityException: Binder invocation to an incorrect interface
        at org.opendatakit.database.service.InternalUserDbInterfaceAidlWrapperImpl.rethrowAlwaysAllowedRemoteException(InternalUserDbInterfaceAidlWrapperImpl.java:138)
        at org.opendatakit.database.service.InternalUserDbInterfaceAidlWrapperImpl.getActiveUser(InternalUserDbInterfaceAidlWrapperImpl.java:155)
        at org.opendatakit.database.service.UserDbInterfaceImpl.getActiveUser(UserDbInterfaceImpl.java:121)
        at org.odk.odknotifications.Activities.MainActivity.getActiveUser(MainActivity.java:259)
        at org.odk.odknotifications.Activities.MainActivity.databaseAvailable(MainActivity.java:282)
        at org.opendatakit.application.CommonApplication.triggerDatabaseEvent(CommonApplication.java:475)
        at org.opendatakit.application.CommonApplication.access$300(CommonApplication.java:47)
        at org.opendatakit.application.CommonApplication$BackgroundServices.doServiceConnected(CommonApplication.java:184)
        at org.opendatakit.application.CommonApplication$BackgroundServices.access$100(CommonApplication.java:75)
        at org.opendatakit.application.CommonApplication$BackgroundServices$2.onServiceConnected(CommonApplication.java:124)
        at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:1780)
        at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1809)
        at android.os.Handler.handleCallback(Handler.java:790)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:187)
        at android.app.ActivityThread.main(ActivityThread.java:7025)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:514)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:888)
```
I can't figure out how it's actually accessing the database? I think some error is there.

**Comment by DreamyPhobic** [comment]
@jbeorse and @linl33 help, please?

**Comment by linl33** [comment]
The first error is because `ODK Services` is used to provide the database service for `bat`. There should be a check for `ODK Services` like the one in [Tables](https://github.com/opendatakit/tables) or [Survey](https://github.com/opendatakit/survey).
`bat` initializes its Firebase instance in `MainActivity.java`. 

The second error is because `bat` is currently only compatible with the 2.x versions. See #7

**Comment by linl33** [comment]
Fixed in opendatakit/skunkworks-bat#26
